### PR TITLE
[MW] Mastery correction, small bugs

### PR DESCRIPTION
**Pull request by abelito75**
Should have broken up this into more pull requests most likely but shrug. Fixed mastery within 2ish% some fights its lower some its higher. Its a dirty fix but it works. The issue is that we can miss some casts and count some twice but we are always count the right number now. aka no more 2 env -> 8 gusts hit from env. It will always be 2 env -> 2-4 gusts procs depending on ef buff. This is a quick fix that works and could be slightly improved if you have a big brain idea. 

Most of the other things in here are removing old dead code/icon changes/tooltip changes

Didn't put anything in the change log as imo we don't want people to see we had issues before

### CODE REVIEW

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js [hunk 1]
```diff
@@ -24,15 +23,32 @@ class EnvelopingMists extends Analyzer {
...
+    if (this.selectedCombatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id)) {
+      this.EVM_HEALING_INCREASE=.4;
+    }
+  }
+
+  EVM_HEALING_INCREASE = 0.3;//check for mistwrap
   healingIncrease = 0;
   gustsHealing = 0;
   lastCastTarget = null;
+  countForGusts = false;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Is this variable needed somewhere else?

File: src/parser/monk/mistweaver/modules/spells/Vivify.js [hunk 2]
```diff
@@ -27,20 +27,32 @@ class Vivify extends Analyzer {
   gustsHealing = 0;
   lastCastTarget = null;
   remDuringManaTea = 0;
+  countForGusts = false;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/Vivify.js]: Is this variable used anywhere else?

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js [hunk 3]
```diff
@@ -24,15 +23,32 @@ class EnvelopingMists extends Analyzer {
     combatants: Combatants,
   };
 
+  constructor(...args) {
+    super(...args);
+    if (this.selectedCombatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id)) {
+      this.EVM_HEALING_INCREASE=.4;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: 2 Style issues here:

1. Use camel case for the variable name, eg. `evmHealingIncrease`
2. Space the statement out, eg `this.evmHealingIncrease = .4;

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js [hunk 4]
```diff
@@ -24,15 +23,32 @@ class EnvelopingMists extends Analyzer {
...
   on_byPlayer_cast(event) {
     const spellId = event.ability.guid;
-    if (SPELLS.ENVELOPING_MIST.id !== spellId) {
+
+    if (SPELLS.ENVELOPING_MIST.id !== spellId) {//bail early if not the right spell
       return;
     }
+    if (this.combatants.players[event.targetID]) {
+      if (this.combatants.players[event.targetID].hasBuff(SPELLS.ESSENCE_FONT_BUFF.id, event.timestamp, 0, 0) === true) {
+        this.numberToCount++;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Another style item, but we have chosen to use `+= 1` to increment. So just rework this is `this.numberToCount += 1;`

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js [hunk 5]
```diff
@@ -24,15 +23,32 @@ class EnvelopingMists extends Analyzer {
...
+      }
+    }
+    this.numberToCount++;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Another style item, but we have chosen to use `+= 1` to increment. So just rework this is `this.numberToCount += 1;`

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js [hunk 6]
```diff
@@ -45,14 +61,15 @@ class EnvelopingMists extends Analyzer {
       return;
     }
 
-    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === event.targetID)) {
+    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === event.targetID) && this.numberToCount >0) {
       this.gustProc += 1;
       this.gustsHealing += (event.amount || 0) + (event.absorbed || 0);
+      this.numberToCount--;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Another style item, but we have chosen to use `-= 1` to decrement. So just rework this is `this.numberToCount -= 1;`

File: src/parser/monk/mistweaver/modules/spells/RenewingMist.js [hunk 7]
```diff
@@ -15,20 +18,31 @@ class RenewingMist extends Analyzer {
...
 
   on_byPlayer_cast(event) {
     const spellId = event.ability.guid;
+
     if (SPELLS.RENEWING_MIST.id !== spellId) {
       return;
+    } 
+    if (this.combatants.players[event.targetID]) {
+      if (this.combatants.players[event.targetID].hasBuff(SPELLS.ESSENCE_FONT_BUFF.id, event.timestamp, 0, 0) === true) {
+        this.numberToCount++;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/RenewingMist.js]: Another style item, but we have chosen to use `+= 1` to increment. So just rework this is `this.numberToCount += 1;`

File: src/parser/monk/mistweaver/modules/spells/RenewingMist.js [hunk 8]
```diff
@@ -15,20 +18,31 @@ class RenewingMist extends Analyzer {
...
+      }
     }
+
+    this.numberToCount++;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/RenewingMist.js]: Another style item, but we have chosen to use `+= 1` to increment. So just rework this is `this.numberToCount += 1;`

File: src/parser/monk/mistweaver/modules/spells/RenewingMist.js [hunk 9]
```diff
@@ -15,20 +18,31 @@ class RenewingMist extends Analyzer {
...
     this.lastCastTarget = event.targetID;
   }
 
   on_byPlayer_heal(event) {
     const spellId = event.ability.guid;
 
-    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === event.targetID)) {
+    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === event.targetID) && this.numberToCount>0) {
       this.gustsHealing += (event.amount || 0) + (event.absorbed || 0);
+      this.numberToCount--;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/RenewingMist.js]: Another style item, but we have chosen to use `-= 1` to decrement. So just rework this is `this.numberToCount -= 1;`

File: src/parser/monk/mistweaver/modules/spells/Vivify.js [hunk 10]
```diff
@@ -27,20 +27,32 @@ class Vivify extends Analyzer {
...
 
   on_byPlayer_cast(event) {
     const spellId = event.ability.guid;
+
     if (SPELLS.VIVIFY.id !== spellId) {
       return;
     }
+    if (this.combatants.players[event.targetID]) {
+      if (this.combatants.players[event.targetID].hasBuff(SPELLS.ESSENCE_FONT_BUFF.id, event.timestamp, 0, 0) === true) {
+        this.numberToCount++;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/Vivify.js]: Another style item, but we have chosen to use `+= 1` to increment. So just rework this is `this.numberToCount += 1;`

File: src/parser/monk/mistweaver/modules/spells/Vivify.js [hunk 11]
```diff
@@ -27,20 +27,32 @@ class Vivify extends Analyzer {
...
+      }
+    }
+
+    this.numberToCount++;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/Vivify.js]: Another style item, but we have chosen to use `+= 1` to increment. So just rework this is `this.numberToCount += 1;`

File: src/parser/monk/mistweaver/modules/spells/Vivify.js [hunk 12]
```diff
@@ -27,20 +27,32 @@ class Vivify extends Analyzer {
...
     this.lastCastTarget = event.targetID;
   }
 
   on_byPlayer_heal(event) {
     const spellId = event.ability.guid;
 
-    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === event.targetID)) {
+    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === event.targetID) && this.numberToCount > 0) {
       this.gustsHealing += (event.amount || 0) + (event.absorbed || 0);
+      this.numberToCount--;
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/Vivify.js]: Another style item, but we have chosen to use `-= 1` to decrement. So just rework this is `this.numberToCount -= 1;`

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js [hunk 13]
```diff
@@ -45,14 +61,15 @@ class EnvelopingMists extends Analyzer {
...
     }
 
     if (this.combatants.players[targetId]) {
       if (this.combatants.players[targetId].hasBuff(SPELLS.ENVELOPING_MIST.id, event.timestamp, 0, 0) === true) {
-        this.healingIncrease += calculateEffectiveHealing(event, EVM_HEALING_INCREASE);
+        this.healingIncrease += calculateEffectiveHealing(event, this.EVM_HEALING_INCREASE);
```
Comment (anom0ly) [inline src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Same stylistic item as above. Just use camel case for variables, and full capitalization for constants. This should be `evmHealingIncrease`

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js (hunk 1 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: This is an artifact of other attempts to fix the mastery issue

File: src/parser/monk/mistweaver/modules/spells/Vivify.js (hunk 2 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/Vivify.js]: This is an artifact of other attempts to fix the mastery issue

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js [hunk 14]
```diff
@@ -24,15 +23,32 @@ class EnvelopingMists extends Analyzer {
...
   };
 
+  constructor(...args) {
+    super(...args);
+    if (this.selectedCombatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id)) {
+      this.EVM_HEALING_INCREASE=.4;
+    }
+  }
+
+  EVM_HEALING_INCREASE = 0.3;//check for mistwrap
```
[the remark this answers is not in the document]
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: I was just reusing the old variable name but I'll change it

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js (hunk 3 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: 1. I was just reusing the old variable name but I'll change it
2. I'm going to switch it to a ternary but will make the spacing nice.

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js (hunk 4 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js (hunk 5 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js (hunk 6 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/RenewingMist.js (hunk 7 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/RenewingMist.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/RenewingMist.js (hunk 8 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/RenewingMist.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/Vivify.js (hunk 10 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/Vivify.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/Vivify.js (hunk 11 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/Vivify.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/Vivify.js (hunk 12 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/Vivify.js]: Will do Mr.BossMan

File: src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js (hunk 13 above)
Comment (abelito75) [reply src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js]: I was just reusing the old variable name but I'll change it

**Pull request merged**